File: package/main/src/String/constantCase.ts
import { words } from "./words";

/**
 * Converts a string to CONSTANT_CASE.
 * @param {string} string_ - Input string
 * @returns {string} CONSTANT_CASE string
 * @example
 * constantCase("helloWorld"); // "HELLO_WORLD"
 * constantCase("Hello World"); // "HELLO_WORLD"
 */
export const constantCase = (string_: string): string =>
  words(string_)
    .map((word) => word.toUpperCase())
    .join("_");

File: package/umt_python/src/string/constant_case.py
from .words import words


def constant_case(string_: str) -> str:
    """
    Converts a string to CONSTANT_CASE.

    Args:
        string_: Input string.

    Returns:
        CONSTANT_CASE string.

    Example:
        >>> constant_case("helloWorld")
        'HELLO_WORLD'
        >>> constant_case("Hello World")
        'HELLO_WORLD'
        >>> constant_case("foo-bar-baz")
        'FOO_BAR_BAZ'
        >>> constant_case("")
        ''
    """
    return "_".join(word.upper() for word in words(string_))
